### PR TITLE
Add ARM64 encoding for group IF_SVE_CK_2A

### PR DESCRIPTION
**Pull request by snickolls-arm**
Adds encoding for `punpkhi`/`punpklo` instructions.

Matching capstone:
```
punpkhi p1.h, p7.b
punpklo p5.h, p3.b
```
Contributing towards #94549.

### CODE REVIEW

**Comment by snickolls-arm** [comment]
@a74nh @kunalspathak @dotnet/arm64-contrib

File: src/coreclr/jit/emitarm64.cpp [hunk 1]
```diff
@@ -21017,6 +21036,11 @@ void emitter::emitDispInsHelp(
             emitDispSveReg(id->idReg3(), INS_OPTS_SCALABLE_D, false); // mmmmm
             break;
```
Comment (amanasifkhalid) [inline src/coreclr/jit/emitarm64.cpp:21038]: Could you please add a comment above this case describing the encoding format? e.g. `// <Pd>.H, <Pn>.B`

**Review by amanasifkhalid — APPROVED** [review]
Aside from the missing comment, LGTM. Thank you!

**Review by kunalspathak — APPROVED** [review]
LGTM

File: src/coreclr/jit/emitarm64.cpp (hunk 1 above)
Comment (kunalspathak) [reply src/coreclr/jit/emitarm64.cpp:21038]: +1

**Comment by kunalspathak** [comment]
failures are https://github.com/dotnet/runtime/issues/97985

**Pull request merged**